nucliadb/src/nucliadb/search/search/rerankers.py

```diff
@@ -120,18 +133,32 @@ def needs_extra_results(self) -> bool:
     def items_needed(self, requested: int, shards: int = 1) -> int:
         return requested
 
-    async def rerank(self, kbid: str, query: str, items: list[RerankableItem]) -> list[RankedItem]:
-        # TODO: not implemented yet, as the reranking code is coupled with
-        # hydration and response composition in find_merge.py
-        reranked = []
+    async def rerank(self, items: list[RerankableItem], options: RerankingOptions) -> list[RankedItem]:
+        """Given a list of rerankable items, boost matches that appear multiple
+        times. The returned list can be smaller than the initial, as repeated
+        matches are deduplicated.
+        """
+        reranked_by_id = {}
         for item in items:
-            reranked.append(
-                RankedItem(
+            if item.id not in reranked_by_id:
+                reranked_by_id[item.id] = RankedItem(
                     id=item.id,
                     score=item.score,
                     score_type=item.score_type,
                 )
-            )
+            else:
+                # it's a mutiple match, boost the score
+                if reranked_by_id[item.id].score < item.score:
+                    # previous implementation noted that we are using vector
+                    # score x2 when we find a multiple match. However, this may
+                    # not be true, as the same paragraph could come in any
+                    # position in the rank fusioned result list
+                    reranked_by_id[item.id].score = item.score * 2
+
```

Comment:
isn't it easier to fix now? do you prefer to fix it in a different pr?

Reply:
What do you want to fix? I'm not sure doing a x2 in one or another is better. IMO we'll have to rethink this weird boosting